core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -2500,6 +2479,74 @@ class GroupCoordinatorTest {
     assertEquals(Errors.ILLEGAL_GENERATION, commitOffsetResult(tp))
   }
 
+  @Test
+  def testTxnCommitOffsetWithFencedInstanceId(): Unit = {
```

Comment:
I am not sure if I understand this test. (1) why do we test with a static member? Fencing could also happen for non-static members now? (2) Why does the member get fenced at all (it seems that `staticMembersJoinAndRebalance` should successfully add two members to the consumer group?) (3) Why do we no pass a `groupInstanceId` into `commitTransactionalOffsets()` on the first call, as we use static group membership?

Reply:
I will make this test more explicit by adding the definition for member.id.